apps/mobile/src/components/WelcomeNewUserOnboarding.tsx

```diff
@@ -0,0 +1,158 @@
+import colors from 'shared/theme/colors';
+
+import { GLogo } from '~/navigation/MainTabNavigator/GLogo';
+import { NotificationsIcon } from '~/navigation/MainTabNavigator/NotificationsIcon';
+import { PostIcon } from '~/navigation/MainTabNavigator/PostIcon';
+import { SearchIcon } from '~/navigation/MainTabNavigator/SearchIcon';
+import { LazyAccountTabItem } from '~/navigation/MainTabNavigator/TabBar';
+
+import { GalleryTouchableOpacity } from './GalleryTouchableOpacity';
+import { Typography } from './Typography';
+import { WelcomeNewUser } from './WelcomeNewUser';
+
+type Props = {
+  username: string;
+};
+
+export function WelcomeNewUserOnboarding({ username }: Props) {
+  // Step 1: Welcome message
+  // Step 2: Post message
+  // Step 3: Profile message
+  const [step, setStep] = useState(1);
+
+  const { bottom } = useSafeAreaInsets();
+  const hasSafeAreaIntersection = bottom !== 0;
+
+  const nextStep = useCallback(() => {
+    if (step === 3) {
+      setStep(0);
+      return;
+    }
+
+    if (step === 1) {
+      // Wait for the bottom sheet to dismiss before moving to the next step
+      setTimeout(() => {
+        setStep((prevStep) => prevStep + 1);
+      }, 400);
+      return;
+    }
+
+    setStep((prevStep) => prevStep + 1);
+  }, [step]);
+
+  if (step > 3 || step === 0) {
+    return null;
+  }
+
+  if (step === 1) {
+    return <WelcomeNewUser username={username} onContinue={nextStep} />;
+  }
+
+  return (
+    <BlurView intensity={4} className="absolute h-full w-full top-0 bg-black/50 z-30">
+      <GalleryTouchableOpacity
+        className="relative h-full w-full"
+        onPress={nextStep}
+        eventElementId="Welcome New User Tutorial"
+        eventName="Welcome New User Tutorial Clicked"
+        eventContext={contexts.Onboarding}
+        properties={{
+          step: step === 2 ? 'Post' : 'Profile',
+        }}
+      >
+        {step === 2 && (
+          <Toast message="Tap here to post an item from your collection" step={step} />
+        )}
+
+        {step === 3 && <Toast message="Tap here to set up your Profile" step={step} />}
+
+        <View
+          style={hasSafeAreaIntersection ? { paddingBottom: bottom } : { paddingBottom: 12 }}
+          className="w-full bottom-0 bg-white dark:bg-black-900 absolute flex flex-row items-center justify-evenly border-t border-porcelain dark:border-black-600"
+        >
+          <TabItem onPress={nextStep} isFocused>
+            <GLogo />
+          </TabItem>
+          <TabItem onPress={nextStep}>
+            <SearchIcon />
+          </TabItem>
+          <TabItem active={step === 2} onPress={nextStep}>
+            <PostIcon color={colors.black['800']} />
+          </TabItem>
+          <TabItem onPress={nextStep}>
+            <NotificationsIcon />
+          </TabItem>
+          <TabItem active={step === 3} onPress={nextStep}>
+            <LazyAccountTabItem />
+          </TabItem>
+        </View>
```

Comment:
A little bit hacky to keep display the bottom navbar while maintaining the full screen blur, added mock tab bar to make it looks real lol

Reply:
brilliant!